### PR TITLE
POCONC-161: Fixes and improvements to oncology diagnosis and medicati…

### PR DESCRIPTION
**Pull request by denniskigen**
Fixes and improvements to the oncology diagnosis and medication history displays (#1080):

- Show 'No diagnosis history records available' alert consistently when no diagnosis history records are found for an oncology patient.
- Show 'No medication change history records available' alert consistently when no medication change records are found for an oncology patient.

### CODE REVIEW

File: src/app/patient-dashboard/oncology/diagnosis-history/oncology-diagnosis-history.component.ts [hunk 1]
```diff
@@ -40,32 +41,38 @@ export class OncologyDiagnosisHistoryComponent implements OnInit, OnDestroy {
...
-          message: 'error fetching patient'
-        });
+      if (patient) {
+        this.patient = patient;
+        this.patientUuid = this.patient.person.uuid;
+        this.loadOncologyDiagnosisHistory();
+      }
+    }, (err) => {
+      this.loadingSummary = false;
+      console.error('Error fetching patient: ', err);
```
Comment (madvincy) [inline src/app/patient-dashboard/oncology/diagnosis-history/oncology-diagnosis-history.component.ts]: is there a reason to have console.error

File: src/app/patient-dashboard/oncology/medication-history/oncology-medication-history.component.ts [hunk 2]
```diff
@@ -40,33 +41,38 @@ export class OncologyMedicationHistoryComponent implements OnInit, OnDestroy {
...
-          id: 'patient',
-          message: 'error fetching patient'
-        });
+      if (patient) {
+        this.patient = patient;
+        this.patientUuid = this.patient.person.uuid;
+        this.loadOncologyMedicationHistory();
+      }
+    }, (err) => {
+      console.log(err);
```
Comment (madvincy) [inline src/app/patient-dashboard/oncology/medication-history/oncology-medication-history.component.ts]: same here

File: src/app/patient-dashboard/oncology/medication-history/oncology-medication-history.component.ts [hunk 3]
```diff
@@ -40,33 +41,38 @@ export class OncologyMedicationHistoryComponent implements OnInit, OnDestroy {
...
+      .subscribe((summary) => {
+        this.loadingSummary = false;
+        this.summaryLoaded = true;
+        if (summary.length) {
+          this.medicalChanges = summary;
+          this.hasData = true;
+        }
+      }, (error) => {
+        this.loadingSummary = false;
+        console.log(error);
```
Comment (madvincy) [inline src/app/patient-dashboard/oncology/medication-history/oncology-medication-history.component.ts]: same here

**Review by madvincy — CHANGES_REQUESTED** [review]

**Review by madvincy — COMMENTED** [review]
Check on the changes

File: src/app/patient-dashboard/oncology/medication-history/oncology-medication-history.component.ts (hunk 3 above)
Comment (denniskigen) [reply src/app/patient-dashboard/oncology/medication-history/oncology-medication-history.component.ts]: Corrected.

File: src/app/patient-dashboard/oncology/medication-history/oncology-medication-history.component.ts (hunk 2 above)
Comment (denniskigen) [reply src/app/patient-dashboard/oncology/medication-history/oncology-medication-history.component.ts]: Corrected.

File: src/app/patient-dashboard/oncology/diagnosis-history/oncology-diagnosis-history.component.ts (hunk 1 above)
Comment (denniskigen) [reply src/app/patient-dashboard/oncology/diagnosis-history/oncology-diagnosis-history.component.ts]: No, it shouldn't be here.

**Review by madvincy — APPROVED** [review]

**Review by maikofelix47 — APPROVED** [review]

**Pull request merged**